### PR TITLE
Facepile mergestyles 2

### PR DESCRIPTION
**Pull request by bennettclark**
#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

Converts SCSS to MergeStyles in Facepile component.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/__snapshots__/Facepile.AddFace.Example.tsx.shot [hunk 1]
```diff
@@ -35,18 +59,22 @@ exports[`Component Examples renders Facepile.AddFace.Example.tsx correctly 1`] =
             font-size: 14px;
             font-weight: 400;
             outline: transparent;
-            padding-bottom: 0;
-            padding-left: 16px;
-            padding-right: 16px;
-            padding-top: 0;
+            padding-bottom: 0px;
+            padding-left: 0px;
+            padding-right: 0px;
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/__snapshots__/Facepile.AddFace.Example.tsx.shot]: These changes here don't look right from a pure sass->getStyles conversion

File: packages/office-ui-fabric-react/src/components/Facepile/Facepile.styles.ts [hunk 2]
```diff
@@ -1,17 +1,154 @@
...
       className
-    ]
+    ],
+    addButton: [
+      classNames.addButton,
+      getFocusStyle(theme, -1),
+      ItemButtonStyles,
+      {
+        selectors: {
+          'button&': {
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/Facepile/Facepile.styles.ts]: >button& [](start = 11, length = 7)

this looks wrong to me... If this is intended, could you explain what you're doing here?

File: packages/office-ui-fabric-react/src/components/Facepile/Facepile.styles.ts [hunk 3]
```diff
@@ -1,17 +1,154 @@
...
+  const { palette, fonts } = theme;
+  const classNames = getGlobalClassNames(GlobalClassNames, theme);
+
+  const ItemButtonStyles: IStyle = {
+    textAlign: 'center',
+    padding: 0,
+    borderRadius: '50%',
+    verticalAlign: 'top',
+    selectors: {
+      'button&': {
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Facepile/Facepile.styles.ts]: is there a need for this specificity? (I'm not completely sure.)

File: packages/office-ui-fabric-react/src/components/Facepile/Facepile.styles.ts (hunk 2 above)
Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/Facepile/Facepile.styles.ts]: (This could have been trying to preserve something from original styling.

Comment (bennettclark) [reply packages/office-ui-fabric-react/src/components/Facepile/Facepile.styles.ts]: The original styling https://github.com/OfficeDev/office-ui-fabric-react/blob/b8ca29ca023b038681daa81b51bb52c927ca1545/packages/office-ui-fabric-react/src/components/Facepile/Facepile.scss#L29  contained styles for both
```
.itemButton {
...
}

button.itemButton {
...
}
```
The `'button&:'` selector here is replicating that.

File: packages/office-ui-fabric-react/src/components/Facepile/Facepile.styles.ts (hunk 3 above)
Comment (bennettclark) [reply packages/office-ui-fabric-react/src/components/Facepile/Facepile.styles.ts]: There are divs with the `ms-Facepile-itemButton` class as well as buttons with the `ms-Facepile-itemButton` class. Only the buttons with that class get these `'button&'` styles applied. This is just replicating the styles that were previously in the scss file and it appears to be working as intended.

File: packages/office-ui-fabric-react/src/components/__snapshots__/Facepile.AddFace.Example.tsx.shot (hunk 1 above)
Comment (bennettclark) [reply packages/office-ui-fabric-react/src/components/__snapshots__/Facepile.AddFace.Example.tsx.shot]: @cliffkoh This seems to match the previous styles: https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/Facepile/Facepile.scss#L32 I'm not sure why the test registered a change, but padding: 0 seems to be correct.

File: packages/office-ui-fabric-react/src/components/__snapshots__/Facepile.Basic.Example.tsx.shot [hunk 4]
```diff
@@ -207,7 +306,7 @@ exports[`Component Examples renders Facepile.Basic.Example.tsx correctly 1`] = `
               {
                 -moz-osx-font-smoothing: grayscale;
                 -webkit-font-smoothing: antialiased;
-                border-radius: 0px;
+                border-radius: 50%;
```
[the remark this answers is not in the document]
Comment (bennettclark) [reply packages/office-ui-fabric-react/src/components/__snapshots__/Facepile.Basic.Example.tsx.shot]: Here's what I have dug up so far.

This does reflect the same styles from the scss. https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/Facepile/Facepile.scss#L33

In the last iteration of this PR, @JasonGore gave this response on this specific issue "Tests will more accurately show JS styling, so that makes sense." https://github.com/OfficeDev/office-ui-fabric-react/pull/4998#pullrequestreview-128170797

Additionally, an inspection in the browser of the current live facepile example https://developer.microsoft.com/en-us/fabric#/components/facepile shows me if you remove the 50% border-radius from the facepile itembuttons, they become square.

[the remark this answers is not in the document]
Comment (cliffkoh) [reply packages/office-ui-fabric-react/src/components/__snapshots__/Facepile.Basic.Example.tsx.shot]: I will help take a look, thanks for the investigation thus far.

[the remark this answers is not in the document]
Comment (cliffkoh) [reply packages/office-ui-fabric-react/src/components/__snapshots__/Facepile.Basic.Example.tsx.shot]: I now understand what is happening. The previous snapshot was capturing a child component's getStyles. In this case, the child component is BaseButton - and the 16px belongs there.

The 16px was being overriden to 0px from CSS, but we didn't capture this in the snapshots in the past. Now that it is fully converted, it is captured :) 

So that's the rationalization process. And the way one would double check on all of these changes is to inspect the element pre-change to ensure it is 0px too (which you've done).

So in other words, this looks good :+1:
